### DISCUSSION TITLE
Feature Request: Database Password Prompt on Launch

### DISCUSSION THREAD

**Discussion by aceuna** [mRemoteNG/mRemoteNG#2650] [Ideas]
Hey everyone

First off, thanks for developing mRemote! it’s a fantastic tool.

We’re planning to use the database integration feature to manage connections across the team. However, we’ve run into a couple of issues.
Our current XML configuration doesn’t import directly, which we can work around by manually recreating connections if needed (But a functioning import would be great).

One key concern, though, is that mRemote doesn’t prompt for a database password on launch when connecting to SQL.
For security reasons, we’d really like an option to require a password prompt for the DB connection each time mRemote is opened.

Another concern is, if someone stores their login data in the tool (in the settings, not in a connection), this can be misused when the tool is opened without a password. Simply click on cancel when opening and create a new file. The password from the settings is then also available here. Is there a solution to this or a best practice approach?

Appreciate any advice on this or if it might be considered for future updates. Thanks in advance!

**Comment by Kvarkas** [comment]
Hello, this soon will be changed, as at the moment I am working on security improvements ;)
In general Idea is as follow: admin user chose action then configuring SQL access (that could be as usualy per static credentials, or per user based) then its just starting admin user will need provide credentials with create rights to build a required structure (they will not be saved) or point to already created structure in DB (following our schema what will be provided) save coonection string (password could be excepted) what will be encrypted and securely stored in local db (that could be also placed on shared folder outside user environmet (and it could be in read mode only, so user could use it but no modify it) so on new run mR will check settings (there is a plan in next release intoduce profiles) and if source of connectors (diferent from now, mR will save only shortcuts like to endpoints where will know what source of date it should use, e.g. SQL for example) once fetch it will ask for password or request it from CyberArk safe (or other external storage solutions) and once connected fetch data requred to make connection. Due security constrains I am trying to solve at this moment mR will not store credentials in a app (like now we first load all from source and keep in memory what its not very secure as practice shows) so on every new connection atemp it will fetch data again so ask password again what could anoing users a lot - so still thinking how to make it more simple. Perhaps will ofer to encrypt them and save temporary only till mR not close, will see
